src/server/vnparse.c

```diff
@@ -1063,4 +1065,3045 @@ vnal_alloc(vnal_t **vp)
+					svrattrl *svrattrl_e;
+
+					snprintf(key_buf, BUF_SIZE, "%s.resources_assigned", noden);
+					if ((svrattrl_e = find_svrattrl_list_entry(r_input->vnodes_data, key_buf, "host,string")) != NULL) {
+						parent_mom = svrattrl_e->al_value;
+					}
+				}
+
+				if (parent_mom == NULL) { /* should not happen */
+
+					if ((err_msg != NULL) && (err_msg_sz > 0)) {
+						snprintf(err_msg, err_msg_sz, "no parent_mom for for %s", noden);
+        					log_event(PBSEVENT_DEBUG2, PBS_EVENTCLASS_JOB, LOG_DEBUG, r_input->jobid, err_msg);
+					}
+					goto release_nodes_exit;
+				}
+
+				strncpy(prev_noden, noden, PBS_MAXNODENAME);
+			} else {
+				/* see if previous entry already matches this */
+				if ((pnode == NULL) || 
+					(strcmp(pnode->nd_name, noden) != 0)) {
+					pnode = find_nodebyname(noden);
+				}
+
+				if (pnode == NULL) { /* should not happen */
+					if ((err_msg != NULL) && (err_msg_sz > 0)) {
+						snprintf(err_msg, err_msg_sz, "no node entry for %s", noden);
+        					log_event(PBSEVENT_DEBUG2, PBS_EVENTCLASS_JOB, LOG_DEBUG, r_input->jobid, err_msg);
+					}
+					goto release_nodes_exit;
+				}
+			}
+#endif
+
+			if (reliable_job_node_find(r_input2->succeeded_mom_list, parent_mom) != NULL) {
+				if (entry > 0) { /* there's something */
+						 /* put in previously */
+					if (have != NULL) {
+						if (pbs_strcat(&have->chunkstr, &have->chunkstr_sz, "+") == NULL)
+							goto release_nodes_exit;
+					}
+				}
+
+				if (((hasprn > 0) && (paren > 0)) ||
+				     ((hasprn == 0) && (paren == 0))) {
+						 /* at the beginning of chunk for current host */
+					if (!found_paren) {
+
+						free(have);
+						have = (resc_limit_t *)malloc(sizeof(resc_limit_t));
+						if (have == NULL) {
+							goto release_nodes_exit;
+						}
+						/* clear "have" counts */
+						resc_limit_init(have);
+
+						if (pbs_strcat(&have->chunkstr, &have->chunkstr_sz, "(") == NULL)
+							goto release_nodes_exit;
+						found_paren = 1;
+
+						if (h_entry > 0) {
+							/* there's already previous exec_host entry */
+							if ((have->h_chunkstr != NULL) &&
+								have->h_chunkstr[0] != '\0') {
+								if (pbs_strcat(&have->h_chunkstr, &have->h_chunkstr_sz, "+") == NULL)
+									goto release_nodes_exit;
+							}
+							if ((have->h2_chunkstr != NULL) &&
+								(have->h2_chunkstr[0] != '\0')) {
+								if (pbs_strcat(&have->h2_chunkstr, &have->h2_chunkstr_sz, "+") == NULL)
+									goto release_nodes_exit;
+							}
+						}
+
+						if (pbs_strcat(&have->h_chunkstr, &have->h_chunkstr_sz, chunk1) == NULL)
+							goto release_nodes_exit;
+						if (pbs_strcat(&have->h2_chunkstr, &have->h2_chunkstr_sz, chunk2) == NULL)
+							goto release_nodes_exit;
+						h_entry++;
+					}
+				}
+				if (have == NULL) {
+					log_err(-1, __func__, "unexpected NULL 'have' value");
+					goto release_nodes_exit;
+				}
+
+				if (!found_paren) {
+					if (pbs_strcat(&have->chunkstr, &have->chunkstr_sz, "(") == NULL)
+			
```

Comment:
Difficult to tell what was changed as opposed to relocated. That's a lot of code.